tests/phpunit/tests/option/registration.php

```diff
@@ -149,4 +149,28 @@ public function test_unregister_setting_removes_default() {
 
 		$this->assertFalse( has_filter( 'default_option_test_default', 'filter_default_option' ) );
 	}
+
+	/**
+	 * @ticket 57674
+	 *
+	 * @covers ::unregister_setting
+	 */
+	public function test_unregister_invalid_setting_does_not_raise_a_php_warning() {
+		$setting = uniqid();
+		set_error_handler(
+			function ( $errno = 0, $errstr = '' ) {
+				$error_types = array(
+					E_ERROR   => 'Fatal',
+					E_WARNING => 'Warning',
+					E_NOTICE  => 'Notice',
+				);
+				$error_type  = isset( $error_types[ $errno ] ) ? $error_types[ $errno ] : 'Unknown';
+				$this->fail( 'PHP ' . $error_type . ': ' . $errstr );
+				return false;
+			}
+		);
```

Comment:
Without this code, a notice or warning causes the test to fail with the following message:
>Trying to access array offset on value of type null

With this code, which prefixes the message with `PHP NOTICE: ` or `PHP WARNING: ` , the test fails with the following message on PHP 7:
>PHP Notice: Trying to access array offset on value of type null

While the error level is nice, I think its introduction should be globally available for all tests such as in the abstract base class.

I think for the patch's commit, the custom error handler can be removed, as the fails without the fix and passes after the fix.

I'm also curious to have a discussion about introducing the above code at test global level. Would be curious of its value in helping to identify the message with its error level.